src/App/Search/AbstractSearchClient.php

```diff
@@ -346,4 +347,124 @@ protected function checkKeywordAgainstSynonyms(string $searchPhrase): string
 
         return $searchPhrase;
     }
+
+    /**
+     * reindexes search index
+     *
+     * @param string $indexName
+     * @param array $analysis
+     *
+     */
+
+    public function reindex(string $indexName, array $analysis)
+    {
+        // call parent constructer so we can get access to elastica client methods
+        parent::__construct();
```

Comment:
Is this required?

Reply:
This is required to connect to elasticsearch otherwise we can't get access to the methods below like getIndex etc.

It basically initiates Elastica/Client.